src/function/table/pragma_detailed_profiling_output.cpp

```diff
@@ -51,17 +61,28 @@ unique_ptr<FunctionOperatorData> PragmaDetailedProfilingOutputInit(ClientContext
 	return make_unique<PragmaDetailedProfilingOutputOperatorData>();
 }
 
-static void SetValue(DataChunk &output, int index, int op_id, int fun_id, string name, double time) {
+// Insert a row into the given datachunk
+static void SetValue(DataChunk &output, int index, int op_id, string annotation, int id, string name, double time,
+                     int sample_counter, int tuple_counter, string extra_info) {
 	output.SetValue(0, index, op_id);
-	output.SetValue(1, index, fun_id);
-	output.SetValue(2, index, move(name));
-	output.SetValue(3, index, time);
+	output.SetValue(1, index, move(annotation));
+	output.SetValue(2, index, id);
+	output.SetValue(3, index, move(name));
+	output.SetValue(4, index, time);
+	output.SetValue(5, index, sample_counter);
+	output.SetValue(6, index, tuple_counter);
+	output.SetValue(7, index, move(extra_info));
 }
 
-static void ExtractExpressions(ChunkCollection &collection, ExpressionInformation &info, DataChunk &chunk, int op_id,
-                               int &fun_id, int sample_tuples_count) {
+static void ExtractFunctions(ChunkCollection &collection, ExpressionInfo &info, DataChunk &chunk, int op_id,
+                             int &fun_id, int sample_tuples_count, int tuples_count) {
 	if (info.hasfunction) {
-		SetValue(chunk, chunk.size(), op_id, fun_id++, info.function_name, double(info.time) / sample_tuples_count);
+		SetValue(chunk, chunk.size(), op_id, "Function", fun_id++, info.function_name,
+		         (info.function_time > 0 && info.function_time < (1UL << 63))
```

Comment:
Same here, can this be done not as a big ternary but as a clear separate step that explains what's going on?